core/src/main/java/org/springframework/security/core/annotation/UniqueSecurityAnnotationScanner.java

```diff
@@ -137,6 +140,64 @@ private MergedAnnotation<A> requireUnique(AnnotatedElement element, List<MergedA
 		};
 	}
 
+	private List<MergedAnnotation<A>> findParameterAnnotations(Method method, Class<?> superOrIfc, Parameter current) {
+		List<MergedAnnotation<A>> directAnnotations = Collections.emptyList();
+		for (Method candidate : superOrIfc.getMethods()) {
+			if (isOverrideFor(method, candidate)) {
+				for (Parameter parameter : candidate.getParameters()) {
+					if (parameter.getName().equals(current.getName())) {
+						directAnnotations = findDirectAnnotations(parameter);
+						if (!directAnnotations.isEmpty()) {
+							return directAnnotations;
+						}
+					}
+				}
+			}
+		}
+		return directAnnotations;
+	}
+
+	private List<MergedAnnotation<A>> findParameterAnnotations(Parameter current) {
+		List<MergedAnnotation<A>> directAnnotations = new ArrayList<>(findDirectAnnotations(current));
+		if (directAnnotations.isEmpty()) {
```

Comment:
To align with other Spring Security code, will you please favor returning early over nesting?

For example, instead of this:

```java
if (directAnnotations.isEmpty()) {
   // do things
}
return directAnnotations;
```

do this:

```java
if (!directAnnotations.isEmpty()) {
    return directAnnotations;
}
// do things
```

This tends to reduce nesting.